src/App.tsx

```diff
@@ -1,18 +1,102 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import { useSearchParams } from 'react-router-dom';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
+
+const items: { title: string, id: number }[] = getNumbers(1, 42)
+  .map(n => {
+    return {
+      title: `Item ${n}`,
+      id: n,
+    };
+  });
+
+interface PagesOnPage<O> {
+  totalPages: number;
+  currentPage: O[];
+}
+
+enum Selects {
+  Three = '3',
+  Five = '5',
+  Ten = '10',
+  Twenty = '20',
+}
+
+const pagination = <T extends unknown>(
+  elements: T[],
+  itemsOnPage: number,
+  selectedPage: number,
+): PagesOnPage<T> => {
+  const totalPages = Math.ceil(elements.length / itemsOnPage);
+  const startIndex = 0 + (itemsOnPage * (selectedPage - 1));
+
+  const currentPage = [...elements].splice(startIndex, itemsOnPage);
+
+  const result: PagesOnPage<T> = {
+    totalPages,
+    currentPage,
+  };
+
+  return result;
+};
 
 export const App: React.FC = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const selectedPage = Number(searchParams.get('page'));
+  const itemsOnPage = Number(searchParams.get('perPage'));
+
+  const { currentPage, totalPages } = pagination(
+    items, itemsOnPage, selectedPage,
+  );
+
+  useEffect(() => {
+    if (searchParams.get('page') === null
+      || searchParams.get('perPage') === null
+    ) {
+      searchParams.set('page', '1');
+      searchParams.set('perPage', '5');
+      setSearchParams(searchParams);
+    }
+  }, [searchParams, setSearchParams]);
+
+  useEffect(() => {
+    if (searchParams.get('page') !== null
+      && searchParams.get('perPage') !== null
+    ) {
+      if (selectedPage < 0 || selectedPage > totalPages
+        || !Object.values(Object(Selects)).includes(String(itemsOnPage))) {
+        searchParams.set('page', '1');
+        searchParams.set('perPage', '5');
+        setSearchParams(searchParams);
+      }
+    }
+  }, [searchParams, setSearchParams]);
+
+  const firstElement = currentPage[0]?.id || 0;
```

Comment:
Why do you use optional chaining here? This is not necessary in this case

Reply:
In this way, I set the default value in purpose to render Info even if a list is empty.